master/pkg/schemas/expconf/hyperparameters_config.go

```diff
@@ -37,14 +63,62 @@ func (h *HyperparameterV0) UnmarshalJSON(data []byte) error {
 		return err
 	}
 	if _, ok := parsed.(map[string]interface{}); ok {
+		parsedMap := parsed.(map[string]interface{})
```

Comment:
try:
```go
if tParsed, ok := parsed.(map[string]interface{}); ok {
    hpType, hasType := parsedMap["type"]
```
the `if _, ok` from before was dropping exactly the value you are looking for.